sql/binlog_crypt_data.cc

```diff
@@ -112,7 +112,8 @@ bool Binlog_crypt_data::init_with_loaded_key(uint sch, const uchar* nonce)
 {
   scheme= sch;
 #ifdef MYSQL_SERVER
-  DBUG_ASSERT(key != NULL && nonce != NULL);
+  DBUG_ASSERT(key != NULL);
+  DBUG_ASSERT(nonce != NULL);
```

Comment:
all the ASSERTs where split - why ?

Reply:
so that if they fail the diagnostics pinpoint the failing part instead of everything (i.e. "C" instead of "A && B && C && D")